pkg/watches/helpers.go

```diff
@@ -207,3 +209,54 @@ func getOpSrcOwnerKey(labels map[string]string) *client.ObjectKey {
 
 	return &client.ObjectKey{Namespace: ownerNamespace, Name: ownerName}
 }
+
+// WatchProxyEvents adds watches for proxy events.
+func WatchProxyEvents(c controller.Controller, client cl.Client, owner string) error {
+	// We only care if the event came from the cluster proxy.
+	pred := predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			if e.Meta.GetName() == proxy.ClusterProxyName {
+				return true
+			}
+			return false
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			// If the cluster operator is ever changed we should kick off an event.
+			// Only check if the old name was "cluster", a kube object cannot have it's
+			// name changed in an update.
+			if e.MetaOld.GetName() == proxy.ClusterProxyName {
+				return true
+			}
+			return false
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
```

Comment:
You have to take `e.DeleteStateUnknown` into account. Please see the watches for child resources for an example.